### PR TITLE
Fix(rllib): Use target_qf_twin head in IQL target prediction

### LINKED ISSUE #64931: [RLlib] IQL twin-Q target uses online qf_twin head instead of frozen target_qf_twin
## Bug

In `DefaultIQLTorchRLModule._forward_train`, the twin-Q branch of `QF_TARGET_PREDS` passes the **online** (trainable) `self.qf_twin` head instead of the frozen `self.target_qf_twin`:

```python
# ray/rllib/algorithms/iql/torch/default_iql_torch_rl_module.py, line 54-55
self._qf_forward_train_helper(
    batch_curr, encoder=self.target_qf_twin_encoder, head=self.qf_twin  # <-- should be self.target_qf_twin
)
```

The Q1 path on line 48 correctly uses `head=self.target_qf` (frozen). The SAC parent's `forward_target` also correctly uses `self.target_qf_twin`.

## Impact

Because `QF_TARGET_PREDS` flows (un-detached) into the value loss, `value_loss.backward(retain_graph=True)` in `IQLTorchLearner.compute_gradients` accumulates spurious gradients onto `qf_twin`'s parameters **after** the correct `critic_twin_loss` gradients have already been stored. This corrupts the twin-Q update and causes `qf_twin_loss` to diverge while `qf_loss` trains normally.

## Expected behavior

Both Q-networks should train symmetrically against the same frozen target predictions. `qf_twin_loss` should decrease alongside `qf_loss`.

## Suggested fix

```diff
- batch_curr, encoder=self.target_qf_twin_encoder, head=self.qf_twin
+ batch_curr, encoder=self.target_qf_twin_encoder, head=self.target_qf_twin
```

### Versions / Dependencies

- Ray: 2.56.0
- Python 3.13
- macOS

### Reproduction script

Run e.g. the IQL example script at https://github.com/ray-project/ray/blob/master/rllib/examples/algorithms/iql/pendulum_iql.py. 

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by mukktinaadh**
Fixes #64931.

In `DefaultIQLTorchRLModule._forward_train`, the twin-Q branch of `QF_TARGET_PREDS` was incorrectly passing the online `self.qf_twin` head instead of the frozen `self.target_qf_twin`. Because `QF_TARGET_PREDS` flows un-detached into the value loss, this caused spurious gradients to accumulate onto `qf_twin`'s parameters after the correct `critic_twin_loss` gradients were already stored, corrupting the twin-Q update and causing loss divergence.

This PR fixes the mathematical correctness by ensuring both Q-networks train symmetrically against the same frozen target predictions.

### CODE REVIEW

**Comment by ArturNiederfahrenhorst** [comment]
I'm changing the test a bit (right now it's not a regression test) and also adding the test to the bazel file.

We can then probably merge soon!
Thanks a lot for the PR!

**Review by ArturNiederfahrenhorst — APPROVED** [review]
